source/agora/node/Registry.d

```diff
@@ -194,7 +197,84 @@ public class NameRegistry: NameRegistryAPI
                  registry_payload.data.addresses, registry_payload.data.public_key);
         validator_map[registry_payload.data.public_key] =
             TypedPayload(payload_type, registry_payload);
-        this.registry_stats.setMetricTo!"registry_record_count"(validator_map.length);
+        this.registry_stats.setMetricTo!"registry_record_count"(validator_map.length + flash_map.length);
+    }
+
+    /***************************************************************************
+
+        Get network addresses corresponding to a flash node that is controlling
+        given public_key
+
+        Params:
+            public_key = the public key that was used to register
+                         the network addresses
+
+        Returns:
+            Network addresses associated with the `public_key`
+
+        API:
+            GET /flash_node
+
+    ***************************************************************************/
+
+    public override const(RegistryPayload) getFlashNode (PublicKey public_key)
+    {
+        if (auto ptr = public_key in flash_map)
+        {
+            log.trace("Successfull GET /flash_node: {} => {}", public_key, *ptr);
+            return (*ptr).payload;
+        }
+        log.trace("Unsuccessfull GET /flash_node: {}", public_key);
+        return RegistryPayload.init;
+    }
+
+    /***************************************************************************
+
+        Register network addresses corresponding to a public key
+
+        Params:
+            registry_payload =
+                the data we want to register with the name registry server
+            channel =
+                a known channel of the registering public key
+
+        API:
+            PUT /flash_node
```

Comment:
The API should be `POST`.
https://restfulapi.net/rest-put-vs-post/

Reply:
I'll do that in a separate commit since `/validator` is `PUT` as well.